Add Apache Solr mixin (#1118)
grafana/jsonnet-libs: apache-solr/README.md

# Apache Solr Mixin

> Apache Solr mixin is a set of configurable Grafana dashboards and alerts.

Reviewer: ```suggestion
The Apache Solr mixin is a set of configurable Grafana dashboards and alerts.
```

Author: updated

The Apache Solr mixin contains the following dashboards:

- Apache Solr cluster overview
- Apache Solr query performance overview
- Apache Solr resource monitoring overview

> - Apache Solr Logs overview

Reviewer 2: ```suggestion
- Apache Solr logs overview
```

Author: updated

- ApacheSolrLowCacheHitRatio
- ApacheSolrHighCoreErrors
- ApacheSolrHighDocumentIndexing

## Apache Solr Cluster Overview

The Apache Solr cluster overview dashboard provides details on cluster, shard, replica and Zookeeper health as well as top core and error metrics.

![Apache Solr Cluster Overview Dashboard 1](https://storage.googleapis.com/grafanalabs-integration-assets/apache-solr/screenshots/apache-solr-cluster-1.png)
![Apache Solr Cluster Overview Dashboard 2](https://storage.googleapis.com/grafanalabs-integration-assets/apache-solr/screenshots/apache-solr-cluster-2.png)

## Apache Solr Query Performance Overview

The Apache Solr query performance overview dashboard provides details on various query load and latency, update handlers, cache, timeout and error metrics.

![Apache Solr Query Performance Overview Dashboard 1](https://storage.googleapis.com/grafanalabs-integration-assets/apache-solr/screenshots/apache-solr-query-performance-1.png)
![Apache Solr Query Performance Overview Dashboard 2](https://storage.googleapis.com/grafanalabs-integration-assets/apache-solr/screenshots/apache-solr-query-performance-2.png)

## Apache Solr Resource Monitoring Overview

> The Apache Solr resource monitoring overview dashboard provides details on connection, threads, core fs usage, JVM and Jetty metrics.

Reviewer 2: ```suggestion
The Apache Solr resource monitoring overview dashboard provides details on connections, threads, core FS usage, as well as JVM and Jetty metrics.
```

Author: updated

Apache Solr logs are enabled by default in the `config.libsonnet` and can be removed by setting `enableLokiLogs` to `false`. Then run `make` again to regenerate the dashboard:

```
{
  _config+:: {
    enableLokiLogs: false,
  },
}
```

In order for the selectors to properly work for system logs ingested into your logs datasource, please also include the matching `job` and `solr_cluster` labels onto the [scrape_configs](https://grafana.com/docs/loki/latest/clients/promtail/configuration/#scrape_configs) as to match the labels for ingested metrics.

```yaml
scrape_configs:
  - job_name: integrations/apache-solr
    static_configs:
      - targets: [localhost]
        labels:
          job: integrations/apache-solr
          instance: '<your-instance-name>'
          solr_cluster: '<your-cluster-name>'
          __path__: /var/log/logs/*.log
```

> ![Apache Solr Logs Overview Dashboard 1](https://storage.googleapis.com/grafanalabs-integration-assets/apache-solr/screenshots/apache-solr-logs-overview.png)

Author: updated

## Alerts Overview

> | Alert                                   | Summary                                                                                                             |

Reviewer: I usually see this alerts section done as a simple list. Is this something we've done before? I don't have a problem with this way as long as it's something that has been accepted before/requested by Grafana.

Author: Yeah, this is how I have been doing it in the past with opensearch, meso, etc

# Apache Solr Mixin

The Apache Solr mixin is a set of configurable Grafana dashboards and alerts.

The Apache Solr mixin contains the following dashboards:

- Apache Solr cluster overview
- Apache Solr query performance overview
- Apache Solr resource monitoring overview
- Apache Solr logs overview

and the following alerts:

- ApacheSolrZookeeperChangeInEnsembleSize
- ApacheSolrHighCPUUsageCritical
- ApacheSolrHighCPUUsageWarning
- ApacheSolrHighHeapMemoryUsageCritical
- ApacheSolrHighHeapMemoryUsageWarning
- ApacheSolrLowCacheHitRatio
- ApacheSolrHighCoreErrors
- ApacheSolrHighDocumentIndexing

## Apache Solr Cluster Overview

The Apache Solr cluster overview dashboard provides details on cluster, shard, replica and Zookeeper health as well as top core and error metrics.

![Apache Solr Cluster Overview Dashboard 1](https://storage.googleapis.com/grafanalabs-integration-assets/apache-solr/screenshots/apache-solr-cluster-1.png)
![Apache Solr Cluster Overview Dashboard 2](https://storage.googleapis.com/grafanalabs-integration-assets/apache-solr/screenshots/apache-solr-cluster-2.png)

> ## Apache Solr Query Performance Overview

Reviewer 3: Another nitpick here, and I've touched on it in the past, but we use "overview" far too frequently.

In this case I would consider it as follows:
- Cluster overview, because it's an overview of the various instances
- Query Performance, **without** overview added since it's a drill down dashboard
- Resource Monitoring, **without** overview since while it does give you an overview you can't really drill down further than this. 

@rgeyer what do you think?

Author: removed overview on Query performance and Resource monitoring dashboards

The Apache Solr query performance overview dashboard provides details on various query load and latency, update handlers, cache, timeout and error metrics.

![Apache Solr Query Performance Overview Dashboard 1](https://storage.googleapis.com/grafanalabs-integration-assets/apache-solr/screenshots/apache-solr-query-performance-1.png)
![Apache Solr Query Performance Overview Dashboard 2](https://storage.googleapis.com/grafanalabs-integration-assets/apache-solr/screenshots/apache-solr-query-performance-2.png)

## Apache Solr Resource Monitoring Overview

The Apache Solr resource monitoring overview dashboard provides details on connections, threads, core FS usage, as well as JVM and Jetty metrics.

![Apache Solr Resource Monitoring Overview Dashboard 1](https://storage.googleapis.com/grafanalabs-integration-assets/apache-solr/screenshots/apache-solr-resource-monitoring-1.png)

> ![Apache Solr Resource Monitoring Overview Dashboard 2](https://storage.googleapis.com/grafanalabs-integration-assets/apache-solr/screenshots/apache-solr-resource-monitoring-2.png)

Reviewer 3: Note this may be increased in terms of amount of screenshots if we change the number of columns.

Author: Updated screenshots
